### ISSUE TITLE
I can't get gist

### ISSUE THREAD

**Issue by cloneconf** [SpinlockLabs/github.dart#288]
I want to get some information about gists, but I'm getting `type '_InternalLinkedHashMap<String, dynamic>' is not a subtype of type 'List<dynamic>?' in type cast` Exception.

To reproduce:

```dart
main() async {
  GitHub github = GitHub();

  var gist = await github.gists.getGist("27fd2f861183802ee9fe6c79c10df644");

  print(gist.createdAt);
}
```

this is the error:
```
Unhandled exception:
type '_InternalLinkedHashMap<String, dynamic>' is not a subtype of type 'List<dynamic>?' in type cast
#0      _$GistFromJson (package:github/src/common/model/gists.g.dart:19:29)
#1      new Gist.fromJson (package:github/src/common/model/gists.dart:49:56)
#2      GistsService.getGist.<anonymous closure> (package:github/src/common/gists_service.dart:50:64)
#3      GitHub.requestJson (package:github/src/common/github.dart:323:34)
<asynchronous suspension>
#4      main (package:my_app/singleton.dart:12:14)
<asynchronous suspension>

Process finished with exit code 255
```

**Comment by robbecker-wf** [comment]
I think I know what this is. I'll investigate and get a fix up.

**Comment by robrbecker** [comment]
So, I have a fix for this, but it will be a breaking change (to be released in 9.0.0). So, I'm going to get a couple other things merged in and then cut a release.

**Comment by cloneconf** [comment]
Thanks @robbecker-wf, i will be waiting for it :heart:
